File: Maxglass.Ecommerce/Maxglass.Ecommerce.DataTransfer/Pedidos/Requests/PedidoCadastroRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Maxglass.Ecommerce.DataTransfer.Pedidos.Requests
{
    public class PedidoCadastroRequest
    {
        public string? Cep { get; set; }
        public string? NumeroEndereco { get; set; }
        public string? ComplementoEndereco { get; set; }
        public int? IdFrete { get; set; }
        public IList<ItemPedidoRequest> itemPedidoRequest { get; set; }
    }
}

File: Maxglass.Ecommerce.Web/src/app/pedidos/models/pedido-cadastro-request.ts
import { ItemPedidoRequest } from './pedido-carrinho-request';

export class PedidoCadastroRequest {
    cep: string;
    numeroEndereco: string;
    complementoEndereco: string;
    idFrete: number;
    itemPedidoRequest: ItemPedidoRequest[];

    constructor(params: Partial<PedidoCadastroRequest>) {
        this.cep = params.cep || null;
        this.numeroEndereco = params.numeroEndereco || null;
        this.complementoEndereco = params.complementoEndereco || null;
        this.idFrete = params.idFrete || null;
        this.itemPedidoRequest = params.itemPedidoRequest || null;
    }
}
